mathesar/models/query.py

```diff
@@ -194,13 +195,50 @@ def initial_columns_described(self):
         )
 
     def _describe_query_column(self, sa_col):
-        return {
-            'alias': sa_col.name,
-            'display_name': self._get_display_name_for_sa_col(sa_col),
+        """
+        Note, has some conditional fields depending on whether the column is an initial column or
+        is generated mid-query (created via a summarization).
+        """
+        alias = sa_col.name
+        initial_db_column = self._get_db_initial_column_by_alias(alias)
+        is_initial_column = initial_db_column is not None
+        output = {
+            'alias': alias,
+            'display_name': self._get_display_name_for_alias(alias),
             'type': sa_col.db_type.id,
             'type_options': sa_col.type_options,
-            'display_options': self._get_display_options_for_sa_col(sa_col),
+            'display_options': self._get_display_options_for_alias(alias),
+            'is_initial_column': is_initial_column,
         }
+        if is_initial_column:
+            initial_dj_column = _get_dj_column_for_initial_db_column(initial_db_column)
+            output = output | {
+                'base_column_name': initial_dj_column.name,
+                'base_table_name': initial_dj_column.table.name,
+            }
```

Comment:
It would be better for these to exist, but be null if the column is not an initial column. This would be more extendable in the long run, and also more consistent at the API.

Otherwise, an API consumer can't tell the spec of what's returned without testing all possible inputs.